File: classes/parte-10-interoperabilidad-y-fronteras-entre-lenguajes/164-elegir-el-lenguaje-correcto-para-cada-componente/implementaciones/c/main.c
#include <stdio.h>
#include <string.h>

int main(void) {
    char tipo[32];
    if (scanf("%31s", tipo) != 1) return 1;
    const char *r;
    if (strcmp(tipo, "sistemas") == 0) r = "Rust";
    else if (strcmp(tipo, "web") == 0) r = "TypeScript";
    else if (strcmp(tipo, "datos") == 0) r = "SQL";
    else r = "Python";
    printf("lenguaje=%s\n", r);
    return 0;
}

File: classes/parte-10-interoperabilidad-y-fronteras-entre-lenguajes/164-elegir-el-lenguaje-correcto-para-cada-componente/implementaciones/rust/main.rs
use std::io::Read;

fn main() {
    let mut s = String::new();
    std::io::stdin().read_to_string(&mut s).unwrap();
    let tipo = s.trim();
    let r = match tipo {
        "sistemas" => "Rust",
        "web" => "TypeScript",
        "datos" => "SQL",
        _ => "Python",
    };
    println!("lenguaje={r}");
}
